### PR TITLE
[dev] PBI AB#66009: Changed Drawer's `onClickOutside` prop to `shouldCloseOnClickOutside`

### PR DESCRIPTION
**Pull request by cbrewer1689**

### CODE REVIEW

File: src/surfaces/drawer/drawer.jsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ const defaultProps = {
     isModal: true,
     maxHeight: undefined,
     maxWidth: undefined,
-    onClickOutside: undefined,
+    shouldCloseOnClickOutside: undefined,
```
Comment (groberts314) [inline src/surfaces/drawer/drawer.jsx:75]: `false` please! ;-)

File: src/surfaces/drawer/drawer.jsx [hunk 2]
```diff
@@ -236,36 +236,36 @@ class Drawer extends React.Component {
...
-        if (onClickOutside) {
+        if (shouldCloseOnClickOutside) {
             document.removeEventListener('click', this.onClickOutside);
         }
 
         if (isFunction(onClose)) {
             onClose(...arguments); // eslint-disable-line prefer-rest-params
         } else {
-            console.warning('Drawer\'s onClose prop is required when using the prop onClickOutside'); // eslint-disable-line no-console
+            console.warning('Drawer\'s onClose prop is required when using the prop shouldCloseOnClickOutside'); // eslint-disable-line no-console
```
Comment (groberts314) [inline src/surfaces/drawer/drawer.jsx:259]: NAB: We should adjust the ESLINT settings to allow `info`, `warning` and `error` maybe, like we did in `church-management` repo.

**Review by groberts314 — CHANGES_REQUESTED** [review]
Looks good!  Just one small change request in default prop value.

File: src/surfaces/drawer/drawer.jsx (hunk 1 above)
Comment (cbrewer1689) [reply src/surfaces/drawer/drawer.jsx]: Fixed.

File: src/surfaces/drawer/drawer.jsx (hunk 2 above)
Comment (cbrewer1689) [reply src/surfaces/drawer/drawer.jsx]: copied from hc and pasted here.

**Review by groberts314 — APPROVED** [review]

File: src/surfaces/drawer/drawer.jsx (hunk 2 above)
Comment (groberts314) [reply src/surfaces/drawer/drawer.jsx]: Nice, thanks!

**Pull request merged**